pkg/manager/manager.go

```diff
@@ -377,14 +383,19 @@ func New(config *rest.Config, options Options) (Manager, error) {
 		}
 	}
 
-	resourceLock, err := options.newResourceLock(leaderConfig, leaderRecorderProvider, leaderelection.Options{
-		LeaderElection:             options.LeaderElection,
-		LeaderElectionResourceLock: options.LeaderElectionResourceLock,
-		LeaderElectionID:           options.LeaderElectionID,
-		LeaderElectionNamespace:    options.LeaderElectionNamespace,
-	})
-	if err != nil {
-		return nil, err
+	var resourceLock resourcelock.Interface
+	if options.LeaderElectionResourceLockInterface != nil && options.LeaderElection {
+		resourceLock = options.LeaderElectionResourceLockInterface
+	} else {
```

Comment:
If `options.LeaderElectionResourceLockInterface` is nil we will go into the `else` clause even if `options.LeaderElection` is false

Reply:
ah nvm, missed the fact that we only construct it her and whether or not we use it is determined elsewhere